### PR TITLE
chore(website): add error component in playground

### LINKED ISSUE #1167: Show generation errors in Playground
#### Reason/Context
Currently, if Modelina fails to generate your models there is no error displayed to the user, it just silently fails.

To solve this issue the following need to change:
- Adapt the UI to have a place to show the error.
- Adapt the request code in the Playground component, so when the request fails, we display this error.
- You probably need to adapt the serverless function to return more factual errors instead of just "Error", see `/src/pages/api/generate`.

Feel free to reach out if you want to pick this up or need guidance or have questions 🙂

### PR DESCRIPTION
**Pull request by Samridhi-98**
**Description**

This PR adds a feat to the modelina website that shows error message on the request failure to generate model.

_**Task Done**_

- [x]  Created `CustomError` component.

- [x] Added new state for error handling in the Playground component. 

- [x] Needs to add logic for hardcode statusCode and error message in the error component.

 
**Related issue(s)**

- Fixes #1167 

**Screenshot**

image

### CODE REVIEW

**Comment by jonaslagoni** [comment]
> [ ] Needs to add logic for hardcode statusCode and error message in the error component.

@Samridhi-98 wanna merge this PR and continue to slowly improve it? Or do you want to include it in this PR?

**Review by jonaslagoni — APPROVED** [review]
LGTM 👍

**Comment by Samridhi-98** [comment]
> > [ ] Needs to add logic for hardcode statusCode and error message in the error component.
> 
> @Samridhi-98 wanna merge this PR and continue to slowly improve it? Or do you want to include it in this PR?

No no, still working on it 😅

**Comment by jonaslagoni** [comment]
Hey @Samridhi-98, is anything blocking you to move forward, or something you wish to discuss?

**Comment by Samridhi-98** [comment]
Hi @jonaslagoni, please review and let me know if any change is needed 🙂

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Looks good in general 👍 

Seems like It does not throw an error when incorrect AsyncAPI documents are provided, for example, try this:
```json
{
  "asyncapi": "2.6.0",
  "info": {
    "title": "Kraken Websockets API",
    "version": "1.8.0",
    "description": "test"
  },
  "channels": {
    "/": {
      "publish": {
        "description": "Send messages to the API",
        "operationId": "processReceivedMessage",
        "message": {
            "$ref": "#/components/messages/noneexisting"
        }
      }
    }
  }
}
```


The only remarks to the design are, It's hard to see the error message, dark on dark, it would be great to have some more contrast. Would that be possible? 🤔 

image

File: modelina-website/src/pages/api/generate.ts [hunk 1]
```diff
@@ -15,57 +15,59 @@ import { getPythonModels } from '@/pages/api/functions/PythonGenerator';
...
+  
+  const modelGenerators: {[key: string]: Function} = {
+    'typescript': getTypeScriptModels,
+    'javascript': getJavaScriptModels,
+    'java': getJavaModels,
+    'go': getGoModels,    
+    'csharp': getCSharpModels,
+    'rust': getRustModels,
+    'python': getPythonModels,
+    'dart': getDartModels,
```
[the remark this answers is not in the document]
Comment (Samridhi-98) [reply modelina-website/src/pages/api/generate.ts:37]: Sure, I'll do the suggested changes. 🙂

**Comment by jonaslagoni** [comment]
@Samridhi-98 still think the bottom text should be a different color, I can barely see it my self 😅 Maybe white?
 
image

Did you test with this input that should give an error? Still does not give anything 🤔 
```
{
  "asyncapi": "2.6.0",
  "info": {
    "title": "Kraken Websockets API",
    "version": "1.8.0",
    "description": "test"
  },
  "channels": {
    "/": {
      "publish": {
        "description": "Send messages to the API",
        "operationId": "processReceivedMessage",
        "message": {
            "$ref": "#/components/messages/noneexisting"
        }
      }
    }
  }
}
```

**Comment by Samridhi-98** [comment]
> @Samridhi-98 still think the bottom text should be a different color, I can barely see it my self 😅 Maybe white?
> 
> image
> 
> Did you test with this input that should give an error? Still does not give anything 🤔
> 
> ```
> {
>   "asyncapi": "2.6.0",
>   "info": {
>     "title": "Kraken Websockets API",
>     "version": "1.8.0",
>     "description": "test"
>   },
>   "channels": {
>     "/": {
>       "publish": {
>         "description": "Send messages to the API",
>         "operationId": "processReceivedMessage",
>         "message": {
>             "$ref": "#/components/messages/noneexisting"
>         }
>       }
>     }
>   }
> }
> ```

Oh! I changed it to red just like image and heading components maybe it overrides the CSS.

**Comment by Samridhi-98** [comment]
Hi @jonaslagoni, I have resolved the color and error message issue by rectifying the override of the paragraph component's text color. I have fixed text color to red,  Please suggest if it would be fine or prefer it to be changed to white.

**Review by jonaslagoni — APPROVED** [review]
LGTM 👍

**Pull request merged**